File: 1015.smallest-integer-divisible-by-k/solution.cpp
class Solution {
public:
    int smallestRepunitDivByK(int K) {
        if (K % 2 == 0 || K % 5 == 0) return -1;
        int tmp = 0;
        for (int i = 1; i < 1000000; i++) {
            tmp = tmp * 10 + 1;
            if (tmp % K == 0) return i;
            if (tmp > K) tmp %= K;
        }
        return -1;
    }
};

File: 1015.smallest-integer-divisible-by-k/solution.kt
class Solution {
    fun smallestRepunitDivByK(k: Int): Int {
        var resid = 1 % k
        var len = 1
        val set = hashSetOf(resid)
        while (resid != 0) {
            resid = (resid * 10 + 1) % k
            len++
            if (resid in set) return -1
            set.add(resid)
        }
        return len
    }
}
